packages/core/test/ClientTest.ts

```diff
@@ -267,28 +268,47 @@ class WoTClientTest {
         expect(value).to.equal(42)
     }
 
-    @test async "write a Property"() {
+    @test async "write a Property with raw readable stream"() {
         //verify the value transmitted
         WoTClientTest.clientFactory.setTrap(
             async (form: Form, content: Content) => {
-                const buffer = await ProtocolHelpers.readStreamFully(content.body);
-                expect(buffer.toString()).to.equal("23");
+                const stream = await ProtocolHelpers.toWoTStream(content.body).getReader().read();
```

Comment:
You don't need to do this transformation again. To read the stream fully you can use [ProtocolHelpers.readStreamFully](https://github.com/relu91/thingweb.node-wot/blob/4c820ab6618905ab5d6179e1279b591041764f14/packages/core/src/protocol-helpers.ts#L159). Besides, after the change, I would rename this variable `valueData` cause it is actually not a stream.